### PR TITLE
Many improvements: multi state events, aligning FPL

### PR DESCRIPTION
**Pull request by akomisarek**

### CODE REVIEW

File: ccd-config-generator/src/main/java/uk/gov/hmcts/ccd/sdk/generator/FixedListGenerator.java [hunk 1]
```diff
@@ -21,13 +26,24 @@ public static void generate(File root, Map<Class, Integer> types) {
       if (c.isEnum()) {
         List<Map<String, Object>> fields = Lists.newArrayList();
 
+        FixedList annotation = (FixedList) c.getAnnotation(FixedList.class);
```
Comment (banderous) [inline ccd-config-generator/src/main/java/uk/gov/hmcts/ccd/sdk/generator/FixedListGenerator.java:29]: Is this requiring all enums to have an annotation to be exported? Could you instead exclude individual types with an ignore annotation?

File: ccd-sdk-types/src/main/java/uk/gov/hmcts/ccd/sdk/types/Event.java [hunk 2]
```diff
@@ -140,9 +157,42 @@ public boolean isForAllStates() {
...
+      return this;
+    }
+    public EventBuilder<T, R, S> forStateTransition(S from, S to) {
+      this.preState = List.of(from.getState());
+      this.postState = to.getState();
+      this.transition = true;
+      return this;
+    }
+
+    public EventBuilder<T, R, S> forStateTransition(List<S> from, S to) {
```
Comment (banderous) [inline ccd-sdk-types/src/main/java/uk/gov/hmcts/ccd/sdk/types/Event.java:178]: How does the builder code that uses this look? Does requiring a List make it verbose?

We could reverse the order and make them it variadic so you can pass in S literals.

Or maybe use a nested builder so we could do:

.forStateTransition().from(State.X, State.Y).to(State.Z)

Maybe something for another PR however.

File: ccd-sdk-types/src/main/java/uk/gov/hmcts/ccd/sdk/types/Field.java [hunk 3]
```diff
@@ -104,10 +104,12 @@
     }
 
     public FieldCollectionBuilder<Type, FieldCollectionBuilder<Parent, Grandparent>> complex() {
-      if (clazz == null) {
-        throw new RuntimeException("Cannot infer type for field: " + id
-            + ". Provide an explicit type.");
-      }
+//      if (clazz == null) {
```
Comment (banderous) [inline ccd-sdk-types/src/main/java/uk/gov/hmcts/ccd/sdk/types/Field.java:107]: When would this happen?

**Comment by akomisarek** [comment]
Thanks Alex for looking into my draft, I hope I will be able to resurrect it!

**Pull request closed without merging**